diff --git a/src/expandableCalendar/index.js b/src/expandableCalendar/index.js
index 4fb12b09556a51314f58f58e81fa51fb6e997e41..954b03e400dccec38f6e97b85e2fb1b9b85c3c74 100644
--- a/src/expandableCalendar/index.js
+++ b/src/expandableCalendar/index.js
@@ -280,9 +280,10 @@ const ExpandableCalendar = (props) => {
                 speed: SPEED,
                 bounciness: BOUNCINESS,
                 useNativeDriver: false
-            }).start();
+            }).start(() => {
+                setPosition(() => _height.current === closedHeight ? Positions.CLOSED : Positions.OPEN);
+            });
             onCalendarToggled?.(_isOpen);
-            setPosition(() => _height.current === closedHeight ? Positions.CLOSED : Positions.OPEN);
             closeHeader(_isOpen);
             resetWeekCalendarOpacity(_isOpen);
         }
