diff --git a/lib/typescript/reanimated2/helperTypes.d.ts b/lib/typescript/reanimated2/helperTypes.d.ts
index f904b0c8a3fdfde54306ca43ae01b566a0e12f14..69ddc6c16114a9c4e153db4253eb96764c6a8347 100644
--- a/lib/typescript/reanimated2/helperTypes.d.ts
+++ b/lib/typescript/reanimated2/helperTypes.d.ts
@@ -8,8 +8,8 @@ type AdaptTransforms<T> = {
     [P in keyof T]: Adaptable<T[P]>;
 };
 type TransformsStyle = Pick<RNTransformsStyle, 'transform'>;
-type TransformStyleTypes = TransformsStyle['transform'] extends readonly (infer T)[] | undefined ? T : never;
-type AnimatedTransform = AdaptTransforms<TransformStyleTypes>[];
+type TransformStyleTypes = Extract<TransformsStyle['transform'], any[]>[number];
+type AnimatedTransform = AdaptTransforms<TransformStyleTypes>[] | string;
 export type AnimateStyle<S> = {
     [K in keyof S]: K extends 'transform' ? AnimatedTransform : S[K] extends ReadonlyArray<any> ? ReadonlyArray<AnimateStyle<S[K][0]>> : S[K] extends object ? AnimateStyle<S[K]> : S[K] extends ColorValue | undefined ? S[K] | number : S[K] | SharedValue<AnimatableValue>;
 };
diff --git a/src/reanimated2/helperTypes.ts b/src/reanimated2/helperTypes.ts
index 60d4bcbd4841ace0a1a2d83cf1bd5b3aa876238c..f5e378b12861688ea67878d566c39c89fe30ed8c 100644
--- a/src/reanimated2/helperTypes.ts
+++ b/src/reanimated2/helperTypes.ts
@@ -31,12 +31,8 @@ type AdaptTransforms<T> = {
 
 type TransformsStyle = Pick<RNTransformsStyle, 'transform'>;
 
-type TransformStyleTypes = TransformsStyle['transform'] extends
-  | readonly (infer T)[]
-  | undefined
-  ? T
-  : never;
-type AnimatedTransform = AdaptTransforms<TransformStyleTypes>[];
+type TransformStyleTypes = Extract<TransformsStyle['transform'], any[]>[number];
+type AnimatedTransform = AdaptTransforms<TransformStyleTypes>[] | string;
 
 export type AnimateStyle<S> = {
   [K in keyof S]: K extends 'transform'
